### ISSUE TITLE
Support multiple simultaneous LR schedulers

### ISSUE THREAD

**Issue by jeanm** [pytorch/pytorch#13022]
## 🚀 Feature

Currently only one `_LRScheduler` can be used with the same optimizer. It would be great to support using multiple ones simultaneously.

## Motivation

It's not uncommon in NLP (see e.g. [here](https://github.com/facebookresearch/InferSent)) to want to use both `StepLR` and `ReduceLROnPlateau`. This isn't possible currently, because `StepLR` calculates learning rate updates based on what was the initial learning rate, and how many epochs have gone by (see `get_lr()`). This is all based on the assumption the learning rate hasn't been touched by anything else.

## Pitch

This can be fixed by having `StepLR`,  `ExponentialLR` (and potentially others) query the optimizer to find out the current learning rate, instead of attempting to work it out what it should be. `ReduceLROnPlateau` already does this in `_reduce_lr()`.

**Comment by fmassa** [comment]
I believe this can be closed with #14010?

**Comment by vincentqb** [comment]
Unfortunately, #14010 had to be reverted due to unexpected behavior described in #20527.

**Comment by vincentqb** [comment]
#24352

**Comment by ajuric** [comment]
Any updates on this?

**Comment by vincentqb** [comment]
A solution was implemented in #26423. Closing this PR, please feel free to re-open if needed.
